File: backend/OlivePlatform/OlivePlatform.Application/Features/OlivePurchases/Responses/OlivePurchaseItemDetailsResponse.cs
using OlivePlatform.Application.Features.Analysis.Responses;

namespace OlivePlatform.Application.Features.OlivePurchases.Responses
{
    public class OlivePurchaseItemDetailsResponse
    {
        public int Id { get; set; }

        public required string Reference { get; set; }

        public int? VarietyId { get; set; }

        public decimal AgreedQuantityKg { get; set; }

        public decimal RemainingQuantityKg { get; set; }

        public decimal PricePerKg { get; set; }

        public int TotalAmount { get; set; }

        public OliveAnalysisInfoResponse? Analysis { get; set; }
    }
}

File: frontend/olive-platform/src/features/olivePurchases/data/responses/OlivePurchaseItemDetailsResponse.ts
import type { OliveAnalysisDetailsResponse } from "../../../analyses/oliveAnalyses/data/responses/OliveAnalysesDetailsResponse";
import type { ProductionStatus } from "../../../production/domain/entities/ProductionStatus";
import type { OliveVarieties } from "../../../shared/entities/OliveVarieties";

export type OlivePurchaseItemDetailsResponse = {
  id: number;

  reference: string;

  purchaseId: number;

  varietyId: OliveVarieties;

  agreedQuantityKg: number;

  pricePerKg: number;

  totalAmount: number | null;

  notes: string | null;

  analysis: OliveAnalysisDetailsResponse | null;
};
